src/components/PendingTxsSnackbar/store.test.js

```diff
@@ -0,0 +1,317 @@
+      expect(store.pendingWithdraws.length).toBe(0);
+      expect(store.pendingTransfers.length).toBe(0);
+      expect(store.pendingResetApproves.length).toBe(0);
+      expect(store.isVisible).toBe(true);
+
+      mockAddTransaction({ txid: 9, status: TransactionStatus.PENDING, type: TransactionType.WITHDRAW });
+      await store.queryPendingTransactions();
+      expect(store.count).toBe(9);
+      expect(store.pendingCreateEvents.length).toBe(2);
+      expect(store.pendingBets.length).toBe(1);
+      expect(store.pendingSetResults.length).toBe(2);
+      expect(store.pendingVotes.length).toBe(2);
+      expect(store.pendingFinalizeResults.length).toBe(1);
+      expect(store.pendingWithdraws.length).toBe(1);
+      expect(store.pendingTransfers.length).toBe(0);
+      expect(store.pendingResetApproves.length).toBe(0);
+      expect(store.isVisible).toBe(true);
+
+      mockAddTransaction({ txid: 10, status: TransactionStatus.PENDING, type: TransactionType.WITHDRAW_ESCROW });
+      await store.queryPendingTransactions();
+      expect(store.count).toBe(10);
+      expect(store.pendingCreateEvents.length).toBe(2);
+      expect(store.pendingBets.length).toBe(1);
+      expect(store.pendingSetResults.length).toBe(2);
+      expect(store.pendingVotes.length).toBe(2);
+      expect(store.pendingFinalizeResults.length).toBe(1);
+      expect(store.pendingWithdraws.length).toBe(2);
+      expect(store.pendingTransfers.length).toBe(0);
+      expect(store.pendingResetApproves.length).toBe(0);
+      expect(store.isVisible).toBe(true);
+
+      mockAddTransaction({ txid: 11, status: TransactionStatus.PENDING, type: TransactionType.TRANSFER });
+      await store.queryPendingTransactions();
+      expect(store.count).toBe(11);
+      expect(store.pendingCreateEvents.length).toBe(2);
+      expect(store.pendingBets.length).toBe(1);
+      expect(store.pendingSetResults.length).toBe(2);
+      expect(store.pendingVotes.length).toBe(2);
+      expect(store.pendingFinalizeResults.length).toBe(1);
+      expect(store.pendingWithdraws.length).toBe(2);
+      expect(store.pendingTransfers.length).toBe(1);
+      expect(store.pendingResetApproves.length).toBe(0);
+      expect(store.isVisible).toBe(true);
+
+      mockAddTransaction({ txid: 12, status: TransactionStatus.PENDING, type: TransactionType.RESET_APPROVE });
+      await store.queryPendingTransactions();
+      expect(store.count).toBe(12);
+      expect(store.pendingCreateEvents.length).toBe(2);
+      expect(store.pendingBets.length).toBe(1);
+      expect(store.pendingSetResults.length).toBe(2);
+      expect(store.pendingVotes.length).toBe(2);
+      expect(store.pendingFinalizeResults.length).toBe(1);
+      expect(store.pendingWithdraws.length).toBe(2);
+      expect(store.pendingTransfers.length).toBe(1);
+      expect(store.pendingResetApproves.length).toBe(1);
+      expect(store.isVisible).toBe(true);
+
+      mockResetTransactionList();
```

Comment:
dont reset (delete) the txs from the list. do what the DB would do, which is change the status of the txs. `PENDING > SUCCESS/FAIL`

Reply:
just make `mockResetTransactionList()` to go through each tx and change the status.

Reply:
Currently the mocked query function is not using params to filter the result. The returned result are set manually.

I was planning to improve the mock query function later to avoid making huge pr, or do you think I should better do it in this pr?